sdcflows/utils/wrangler.py

```diff
@@ -243,47 +258,58 @@ def find_estimators(*, layout, subject, sessions=None, fmapless=True, force_fmap
                        bids_id='auto_00000')]
 
     """
-    from .. import fieldmaps as fm
+    from .misc import create_logger
     from bids.layout import Query
     from bids.exceptions import BIDSEntityError
 
+    # Set up logger (logs only visable if verbose)
+    logger = create_logger('sdcflows.wrangler', level=10 if verbose else 40)
+
     base_entities = {
         "subject": subject,
         "extension": [".nii", ".nii.gz"],
         "scope": "raw",  # Ensure derivatives are not captured
     }
 
     subject_root = Path(layout.root) / f"sub-{subject}"
-    sessions = sessions if sessions else layout.get_sessions(subject=subject)
-
+    sessions = sessions or layout.get_sessions(subject=subject)
     fmapless = fmapless or {}
     if fmapless is True:
         fmapless = {"bold", "dwi"}
 
     estimators = []
 
-    b0_ids = tuple()
+    # Step 1. Use B0FieldIdentifier metadata
+    has_B0FI = False
     with suppress(BIDSEntityError):
         b0_ids = layout.get_B0FieldIdentifiers(**base_entities)
+        has_B0FI = True
+
+    if has_B0FI:
+        logger.debug(
+            "Dataset includes `B0FieldIdentifier` metadata."
+            "Any data missing this metadata will be ignored."
+        )
+        for b0_id in b0_ids:
+            # Found B0FieldIdentifier metadata entries
+            b0_entities = base_entities.copy()
+            b0_entities["B0FieldIdentifier"] = b0_id
+
+            e = fm.FieldmapEstimation([
+                fm.FieldmapFile(fmap.path, metadata=fmap.get_metadata())
+                for fmap in layout.get(**b0_entities)
+            ])
+            _log_debug_estimation(logger, e, layout.root)
+            estimators.append(e)
```

Comment:
```suggestion
    for b0_id in b0_ids:
        # Found B0FieldIdentifier metadata entries
        entities = base_entities.copy()
        entities["B0FieldIdentifier"] = b0_id

        _e = fm.FieldmapEstimation([
            fm.FieldmapFile(fmap.path, metadata=fmap.get_metadata())
            for fmap in layout.get(**entities)
        ])
        estimators.append(_e)
```

Reply:
ditto (no problem if you want to change the `_e` variable name though.

Reply:
why deindent this if it is only relevant if `b0_ids` is not empty

Reply:
I'm not super-strong on this one. That said, IMHO it is better to avoid Python indentations when possible. Also, from a PR point of view, the less lines that *need* a change, the better.

Reply:
I'm gonna go ahead with it, on the basis that I think it makes the logic simpler to follow